src/platform/user/authentication/components/OrganicAdoptionExperimentModal.jsx

```diff
@@ -0,0 +1,104 @@
+import React, { useEffect, useState } from 'react';
+import { VaModal } from '@department-of-veterans-affairs/component-library/dist/react-bindings';
+
+import * as authUtilities from 'platform/user/authentication/utilities';
+import { getQueryParams } from '../utilities';
+import recordEvent from '../../../monitoring/record-event';
+import { externalApplicationsConfig } from '../usip-config';
+
+const OrganicAdoptionExperimentModal = ({ visible = false, onClose }) => {
+  const [useOAuth, setOAuth] = useState();
+  const { OAuth } = getQueryParams();
+  const { OAuthEnabled } = externalApplicationsConfig.default;
+
+  useEffect(
+    () => {
+      setOAuth(OAuthEnabled && OAuth === 'true');
+    },
+    [OAuth, OAuthEnabled],
+  );
+
+  let logingovSingUpLink;
+  // uses logic from CreateAccountLink component
+  async function generateURL() {
+    const url = await authUtilities.signupOrVerify({
+      policy: 'logingov',
+      isLink: true,
+      allowVerification: false,
+      useOAuth,
+    });
```

Comment:
Take this
```javascript
await authUtilities.signupOrVerify({
      policy: 'logingov',
      isLink: true,
      allowVerification: false,
      useOAuth,
    });
``` and just inline it after `onPrimaryButtonClick` function (turn it into an async) on line 58